Extra CR comments from #95621

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #96043

Specifically:
https://github.com/pytorch/pytorch/pull/95621/files/063e44147152f4dd7e51852cf8c679692bd9fd53#r1120306196
https://github.com/pytorch/pytorch/pull/95621#discussion_r1125015510

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

ezyang committed:
Extra CR comments from #95621

Specifically:
https://github.com/pytorch/pytorch/pull/95621/files/063e44147152f4dd7e51852cf8c679692bd9fd53#r1120306196
https://github.com/pytorch/pytorch/pull/95621#discussion_r1125015510

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

[ghstack-poisoned]

ezyang committed:
Update on "Extra CR comments from #95621"


Specifically:
https://github.com/pytorch/pytorch/pull/95621/files/063e44147152f4dd7e51852cf8c679692bd9fd53#r1120306196
https://github.com/pytorch/pytorch/pull/95621#discussion_r1125015510

Signed-off-by: Edward Z. Yang <ezyangmeta.com>

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

ezyang committed:
Update on "Extra CR comments from #95621"


Specifically:
https://github.com/pytorch/pytorch/pull/95621/files/063e44147152f4dd7e51852cf8c679692bd9fd53#r1120306196
https://github.com/pytorch/pytorch/pull/95621#discussion_r1125015510

Signed-off-by: Edward Z. Yang <ezyangmeta.com>

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

albanD commented on torch/_dynamo/utils.py:
```diff
@@ -957,11 +957,10 @@ def same(
-                ref = ref.item()
-            if is_numpy_int_type(res):
-                res = res.item()
+        if relax_numpy_equality and not (
```
Why the `not` here? You actually want to change what happens?

ezyang replied:
The logic you see here in the diff is not what the logic originally was. The ORIGINAL logic was

```
    if relax_numpy_equality:
        ref = ref.item()
```

But if you passed in both ref and res and numpy things, you would error, because ref would get converted to int, but res would not. I sort of hacked this up, but the hack up was wrong.

With the new logic, I more faithfully represent the bias the old implementation had: only ref gets converted from numpy to int, but ONLY if res is not a numpy. I don't allow ref to NOT be numpy and res to be numpy, which seems reasonable to me, because we shouldn't hallucinate numpy integers in the compiled code when the original didn't have any.

albanD replied:
Sounds good! Thanks for the explanation.

ezyang commented:
@pytorchbot merge

ezyang commented:
@pytorchbot merge -r master

ezyang committed:
Update on "Extra CR comments from #95621"


Specifically:
https://github.com/pytorch/pytorch/pull/95621/files/063e44147152f4dd7e51852cf8c679692bd9fd53#r1120306196
https://github.com/pytorch/pytorch/pull/95621#discussion_r1125015510

Signed-off-by: Edward Z. Yang <ezyangmeta.com>

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

ezyang commented:
@pytorchbot merge -f "spurious failures"
